samples: sample.net.sockets.http.server: build failures on many NXP platforms

### Describe the bug

- frdm_k64f/mk64f12
- mimxrt1170_evk@A/mimxrt1176/cm7
- mimxrt1064_evk/mimxrt1064
- mimxrt1050_evk/mimxrt1052/hyperflash

build failure due to

https://github.com/zephyrproject-rtos/zephyr/pull/87269

3ffd1675eb5dc7ca6570d1bfea5f824d0d1a232c is the first bad commit
```
commit 3ffd1675eb5dc7ca6570d1bfea5f824d0d1a232c
Author: Johann Fischer <johann.fischer@nordicsemi.no>
Date:   Mon Nov 25 13:54:50 2024 +0100

    samples: http_server: use new USB device stack
    
    Remove netusb code, use new USB device stack and CDC NCM implementation.
    
    Signed-off-by: Johann Fischer <johann.fischer@nordicsemi.no>

 samples/net/sockets/http_server/CMakeLists.txt     |  5 ++++-
 samples/net/sockets/http_server/Kconfig            |  7 ++++++
 samples/net/sockets/http_server/README.rst         |  2 +-
 .../net/sockets/http_server/overlay-netusb.conf    | 10 ---------
 samples/net/sockets/http_server/overlay-usbd.conf  |  7 ++++++
 samples/net/sockets/http_server/sample.yaml        |  6 +++++
 samples/net/sockets/http_server/src/main.c         | 25 ++++++++++++++++-----
 samples/net/sockets/http_server/src/usb.c          | 26 ----------------------
 .../net/sockets/http_server/usbd_cdc_ncm.overlay   | 12 ++++++++++
 9 files changed, 57 insertions(+), 43 deletions(-)
 delete mode 100644 samples/net/sockets/http_server/overlay-netusb.conf
 create mode 100644 samples/net/sockets/http_server/overlay-usbd.conf
 delete mode 100644 samples/net/sockets/http_server/src/usb.c
 create mode 100644 samples/net/sockets/http_server/usbd_cdc_ncm.overlay
```

### Regression

- [x] This is a regression.

### Steps to reproduce

```
scripts/twister -p frdm_k64f -j 12 -T samples/net/sockets/http_server --build-only -M all
```
or
```
west build -b frdm_k64f samples/net/sockets/http_server
```

### Relevant log output

```shell
-- Zephyr version: 4.2.99 (/home/jenkins/agent/workspace/zephyr_downstream_build_daily/zephyr), build: v4.2.0-10671-g2e824953d5f9[6/437] Preparing syscall dependency handling[7/437] Generating misc/generated/syscalls_subdirs.trigger[8/437] Generating misc/generated/syscalls.json, misc/generated/struct_tags.json[9/437] Generating include/generated/zephyr/syscall_dispatch.c, include/generated/zephyr/syscall_exports_llext.c, syscall_weakdefs_llext.c, include/generated/zephyr/syscall_list.h[10/437] Generating include/generated/zephyr/driver-validation.h[11/437] Generating include/generated/zephyr/kobj-types-enum.h, include/generated/zephyr/otype-to-str.h, include/generated/zephyr/otype-to-size.h[12/437] Generating include/generated/device-api-sections.ld, include/generated/device-api-sections.cmake[13/437] Building C object zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj[14/437] Generating include/generated/zephyr/offsets.h[15/437] Building C object CMakeFiles/app.dir/src/main.c.objFAILED: CMakeFiles/app.dir/src/main.c.obj nstream_build_daily/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wdouble-promotion -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -ftls-model=local-exec -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/home/jenkins/agent/workspace/zephyr_downstream_build_daily/zephyr/samples/net/sockets/http_server=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/jenkins/agent/workspace/zephyr_downstream_build_daily/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/jenkins/agent/workspace/zephyr_downstream_build_daily=WEST_TOPDIR -ffunction-sections -fdata-sections -specs=picolibc.specs -D_POSIX_THREADS -std=c99 -MD -MT CMakeFiles/app.dir/src/main.c.obj -MF CMakeFiles/app.dir/src/main.c.obj.d -o CMakeFiles/app.dir/src/main.c.obj -c /home/jenkins/agent/workspace/zephyr_downstream_build_daily/zephyr/samples/net/sockets/http_server/src/main.c/home/jenkins/agent/workspace/zephyr_downstream_build_daily/zephyr/samples/net/sockets/http_server/src/main.c:22:10: fatal error: sample_usbd.h: No such file or directory 22 | #include <sample_usbd.h> | ^~~~~~~~~~~~~~~compilation terminated.ninja: build stopped: subcommand failed.
```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

_No response_

### Additional Context

_No response_

--- a/samples/net/sockets/http_server/src/main.c
+++ b/samples/net/sockets/http_server/src/main.c
@@ -19,8 +19,11 @@
 #include <zephyr/drivers/led.h>
 #include <zephyr/data/json.h>
 #include <zephyr/sys/util_macro.h>
-#include <sample_usbd.h>
 #include <zephyr/net/net_config.h>
+
+#if CONFIG_USB_DEVICE_STACK_NEXT
+#include <sample_usbd.h>
+#endif
 
 #include "ws.h"
 

